src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -23596,9 +23600,20 @@ Statement* Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
         {
             unsigned tmpNum = inlineInfo->lclTmpNum[lclNum];
 
-            // Is the local used at all?
+            // If the local is used check whether we need to insert explicit zero initialization.
             if (tmpNum != BAD_VAR_NUM)
             {
+                if (!fgVarNeedsExplicitZeroInit(&lvaTable[tmpNum], bbInALoop))
```

Comment:
Nit: please use `lvaGetDesc(varNum)`  instead of `&lvaTable[tmpNum]` here and below.

Reply:
Done.